app/controllers/concerns/api_actionable.rb

```diff
@@ -28,6 +33,24 @@ def check_for_serve_file_action
     serve_file_action.update(lifecycle_stage: 'complete', lifecycle_message: "label: #{file.label} id: #{file.id} mime_type: #{file.attachment.content_type}")
   end
 
+  def handle_custom_action
+    flash[:notice] = @api_namespace.api_form.success_message
+    if @custom_action.present?
+      begin
+        custom_api_action = CustomApiAction.new
+
+        eval("def custom_api_action.run_custom_action(api_action: , api_namespace:, api_resource: , current_visit: , current_user: nil); #{@custom_action.method_definition};end;")
```

Comment:
@alis-khadka @Pralish  we should create a ticket that scrubs method/model definitions coming into Violet Rails of things like: `Subdomain`, `Apartment`, `Tenant`, `switch`, `SomeModel.destroy_all` -- so we can open this upto platform developers on subdomains instead of whole systems (like how we have it now, because of this vulnerability)